### PR TITLE
Javascript solution for Longest Substring Without Repeating Characters

### PR DESCRIPTION
**Pull request by MithunPrabhu777**
## Summary

Javascript solution for Longest Substring Without Repeating Characters

## Checklist

**If you haven't fulfilled the below requirements, your PR won't be reviewed.**

### General 

- [x] Make sure you've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md) 
- [x] Make sure you've followed the template format
- [x] Make sure you've started the app locally and verified all the content and all links (if applicable) must be accessible correctly
- [x] Make sure you've applied LaTex for variables, formulas and time / space complexity instead of using backticks
- [ ] Make sure you've formatted your code well (Code Block is rendered using `<Tabs/>`)
      Below is a C++ example. You can include multiple langs using `<TabItem/>` if necessary
      
````
<Tabs>

<TabItem value="cpp" label="C++">
<SolutionAuthor name="@YOUR_ALIAS"/>

```cpp
// Your code goes here
// Prefer K&R Coding style
```

</TabItem>

</Tabs>
````
- [x] Make sure you've explained your code well (write comments above each line - not inline)
- [x] Make sure you've written your explanation well and it is easy to understand for beginners
- [ ] Make sure you've included Time & Space Complexity Analysis
- [ ] Make sure you've included a meaningful approach name for your solution. Don't leave it blank. e.g. `## Approach 1: Two Pointers`.

### For Tutorial Articles

- [ ] Make sure you've explained your topic using 1 - 3 LC problems rather than using external problems. Solutions must be taken with explanations as well.
- [ ] Make sure you've provided suggested problems at the end using the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [ ] Make sure you've given credits / references if you use external resources. For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).

### CODE REVIEW

File: solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md [hunk 1]
```diff
@@ -116,6 +116,42 @@ class Solution {
...
 </TabItem>
+
+<TabItem value="javascript" label="JavaScript">
+<SolutionAuthor name="@MithunPrabhu777"/>
+
+```javascript
+var lengthOfLongestSubstring = function (s) {
+    const letterCountMap = new Map();
+    
+    let [left,right,max] = [0,0,0];
```
Comment (vigneshshiv) [inline solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md:127]: Leave a space here, between each arguments

File: solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md [hunk 2]
```diff
@@ -116,6 +116,42 @@ class Solution {
 }
 ```
 </TabItem>
+
+<TabItem value="javascript" label="JavaScript">
+<SolutionAuthor name="@MithunPrabhu777"/>
+
+```javascript
+var lengthOfLongestSubstring = function (s) {
```
Comment (vigneshshiv) [inline solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md:124]: Is this code tested in LC ?

File: solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md [hunk 3]
```diff
@@ -116,6 +116,42 @@ class Solution {
...
+    const letterCountMap = new Map();
+    
+    let [left,right,max] = [0,0,0];
+    
+    while(right < s.length){
```
Comment (vigneshshiv) [inline solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md:129]: Too many empty lines, leave empty line wherever it's needed. 
Add comments for each line and why we are doing this operation..

**Review by vigneshshiv — CHANGES_REQUESTED** [review]
Provide comments in code and format it properly.

File: solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md (hunk 2 above)
Comment (MithunPrabhu777) [reply solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md:124]: Yes,this code is tested in leetcode

File: solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md (hunk 1 above)
Comment (MithunPrabhu777) [reply solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md:127]: Drafting changes

File: solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md (hunk 3 above)
Comment (MithunPrabhu777) [reply solutions/0000-0099/0003-longest-substring-without-repeating-characters-medium.md:129]: Drafting changes

**Review by vigneshshiv — APPROVED** [review]
LGTM! @wingkwong please proceed.

**Review by wingkwong — APPROVED** [review]

**Pull request merged**